src/fromager/overrides.py

```diff
@@ -108,12 +109,23 @@ def patches_for_requirement(
     override_name = pkgname_to_override_module(req.name)
     unversioned_patch_dir = patches_dir / override_name
     versioned_patch_dir = patches_dir / f"{override_name}-{version}"
+
+    unversioned_patch_files = list(unversioned_patch_dir.glob("*.patch"))
+    versioned_patch_files = list(versioned_patch_dir.glob("*.patch"))
+
+    # The list of files must exist to be joined to the global patch files
+    if variant:
+        unversioned_patch_files += list(
+            unversioned_patch_dir.glob(f"*.patch.{variant}")
+        )
+        versioned_patch_files += list(versioned_patch_dir.glob(f"*.patch.{variant}"))
```

Comment:
- I don't like the `.variant` suffix. How about a subdirectory?
- use list's `extend()` method to extend the lists.